modules/platforms/dotnet/Apache.Ignite.Core/Impl/Client/Transactions/TransactionsClient.cs

```diff
@@ -0,0 +1,249 @@
+        }
+
+        /** <inheritDoc /> */
+        public TransactionConcurrency DefaultTxConcurrency
+        {
+            get { return _cfg.DefaultTransactionConcurrency; }
+        }
+
+        /** <inheritDoc /> */
+        public TransactionIsolation DefaultTxIsolation
+        {
+            get { return _cfg.DefaultTransactionIsolation; }
+        }
+
+        /** <inheritDoc /> */
+        public TimeSpan DefaultTimeout
+        {
+            get { return _cfg.DefaultTimeout; }
+        }
+
+        /** <inheritDoc /> */
+        public ITransactionClient TxStart()
+        {
+            return TxStart(_cfg.DefaultTransactionConcurrency, _cfg.DefaultTransactionIsolation);
+        }
+
+        /** <inheritDoc /> */
+        public ITransactionClient TxStart(TransactionConcurrency concurrency, TransactionIsolation isolation)
+        {
+            return TxStart(concurrency, isolation, _cfg.DefaultTimeout);
+        }
+
+        /** <inheritDoc /> */
+        public ITransactionClient TxStart(TransactionConcurrency concurrency, 
+            TransactionIsolation isolation,
+            TimeSpan timeout)
+        {
+            return TxStart(concurrency, isolation, timeout, null);
+        }
+
+        private ITransactionClient TxStart(TransactionConcurrency concurrency,
+            TransactionIsolation isolation,
+            TimeSpan timeout,
+            string label)
+        {
+            if (CurrentTx != null)
+            {
+                throw new IgniteClientException("A transaction has already been started by the current thread.");
+            }
+
+            var tx = _ignite.Socket.DoOutInOp(
+                ClientOp.TxStart,
+                ctx =>
+                {
+                    ctx.Writer.WriteByte((byte) concurrency);
+                    ctx.Writer.WriteByte((byte) isolation);
+                    ctx.Writer.WriteTimeSpanAsLong(timeout);
+                    ctx.Writer.WriteString(label);
+                },
+                ctx => new TransactionClient(
+                    ctx.Reader.ReadInt(),
+                    _ignite,
+                    ctx.Socket)
+            );
+
+            _currentTx.Value = tx;
+            return _currentTx.Value;
+        }
+
+        /** <inheritDoc /> */
+        public ITransactionsClient WithLabel(string label)
+        {
+            IgniteArgumentCheck.NotNullOrEmpty(label, "label");
+
+            return new TransactionsClientWithLabel(this, label); 
+        }
+
+        /** <inheritDoc /> */
+        [SuppressMessage("Microsoft.Usage",
+            "CA1816:CallGCSuppressFinalizeCorrectly",
+            Justification = "There is no finalizer.")]
+        public void Dispose()
+        {
+            _currentTx.Dispose();
+            _txManager.Dispose();
+        }
+
+        /// <summary>
+        /// Wrapper for transactions with label.
+        /// </summary>
+        private class TransactionsClientWithLabel : ITransactionsClientInternal
```

Comment:
I think it will be simpler to incorporate label into the `TransactionsClient` class.